libraries/ESP8266WiFi/src/ESP8266WiFiAP.h

```diff
@@ -37,8 +37,8 @@ class ESP8266WiFiAPClass {
 
     public:
 
-        bool softAP(const char* ssid, const char* psk = NULL, int channel = 1, int ssid_hidden = 0, int max_connection = 4);
-        bool softAP(const String& ssid,const String& psk = emptyString,int channel = 1,int ssid_hidden = 0,int max_connection = 4);
+        bool softAP(const char* ssid, const char* psk = NULL, int channel = 1, int ssid_hidden = 0, int max_connection = 4, int beacon_interval = 100);
+        bool softAP(const String& ssid,const String& psk = emptyString,int channel = 1,int ssid_hidden = 0,int max_connection = 4,int beacon_interval = 100);

```

Comment:
[Same comment](https://github.com/esp8266/Arduino/pull/8758#discussion_r1048280597) may apply (to avoid possible breaking change)

```cpp

        bool softAP(const char* ssid, const char* psk = NULL, int channel = 1, int ssid_hidden = 0, int max_connection = 4);
        bool softAP(const String& ssid,const String& psk = emptyString,int channel = 1,int ssid_hidden = 0,int max_connection = 4);
+       bool softAP(const char* ssid, const char* psk, int channel, int ssid_hidden, int max_connection, int beacon_interval);
+       bool softAP(const String& ssid,const String& psk,int channel,int ssid_hidden,int max_connection,int beacon_interval);
```

Reply:
I don't know if I was clear and sorry for that.
I meant it is preferable to do this, in case of subclassing / overloading.
```cpp
        bool softAP(const char* ssid, const char* psk = NULL, int channel = 1, int ssid_hidden = 0, int max_connection = 4)
        {
                return softAP(ssid, psk, channel, hidden, max_connection, 100);
        }

+       bool softAP(const char* ssid, const char* psk, int channel, int ssid_hidden, int max_connection, int beacon_interval);
```

Reply:
I don't believe this matters in our case too too much?

Ambiguous overload is resolved at build time, everything is ok on that part.
If we worry about faulty default method used, arguments are expanded left-to-right so user will magically use the new method after upgrading our version.
If we worry about binary stability and preserving existing function signature in the .o file... idk how does one need to build its software for it to matter :)

Reply:
That's right. What I was saying seems to be only relevant when the method is virtual.